aptos-move/block-executor/src/view.rs

```diff
@@ -279,3 +352,63 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>> TStateView for LatestView<
         self.base_view.get_usage()
     }
 }
+
+impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> ExecutableView
+    for LatestView<'a, T, S, X>
+{
+    type Executable = X;
+    type Key = T::Key;
+
+    fn fetch_module(
+        &self,
+        key: &T::Key,
+    ) -> anyhow::Result<(ExecutableDescriptor, FetchedModule<X>)> {
+        match self.latest_view {
+            ViewMapKind::MultiVersion(map) => map
+                .fetch_code(key, self.txn_idx)
+                .map_or_else(|| self.get_base_module(key), Ok),
+            ViewMapKind::Unsync(map) => map.fetch_code(key).map_or_else(
+                || self.get_base_module(key),
+                |res| {
+                    Ok(match res {
+                        MVCodeOutput::Executable((executable, descriptor)) => {
+                            (descriptor, FetchedModule::Executable(executable))
+                        },
+                        MVCodeOutput::Module((v, hash)) => (
+                            ExecutableDescriptor::Published(hash),
+                            FetchedModule::Blob(
+                                v.extract_raw_bytes().expect("Module can't be deleted"),
+                            ),
+                        ),
+                    })
+                },
+            ),
+        }
+    }
+
+    fn store_executable(
```

Comment:
when do we expect to call this? seems a new interaction between vm and storage

Reply:
gives flexibility on the Move-VM side to decide what to cache, i.e. we may not want to cache (currently load into loader) a dependency executable that was only created for verification purposes (of a different module/executable). In the future when we improve those behaviors, maybe it can be made intelligently automatic